other/audit-secret-token-type/audit-secret-token-type.yaml

```diff
@@ -0,0 +1,42 @@
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: audit-service-account-token-secrets
+  annotations:
+    policies.kyverno.io/title: Audit Service Account Token Secrets
+    policies.kyverno.io/category: Sample
+    policies.kyverno.io/severity: medium
+    policies.kyverno.io/subject: Secret
+    policies.kyverno.io/minversion: 1.6.0
+    policies.kyverno.io/description: >-
+      This policy audits for secrets of type 'kubernetes.io/service-account-token'.
+      In version v1.24, more enhancements have been introduced, affecting the behavior of 
+      service accounts and their secrets.In previous versions, when you create a service account, 
+      it automatically creates a secret resource associated with the service account.In version
+      v1.24 when you create a service account, secrets are no longer automatically created. 
+      To create a token,refer to the documentation for more details  
+      https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/ and
+      https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/
+spec:
+  validationFailureAction: audit
+  background: true
+  rules:
+  - name: validate-service-account-token-secrets
+    match:
+      any:
+      - resources:
+          kinds:
+          - Secret
+          namePattern: ".*" 
+          apiGroups: [""]
+          apiVersions: ["v1"]
+          resourceNames: [""] 
```

Comment:
Where do you get these from, ChatGPT? This isn't a valid policy at all and you would have known that if you tried to even create it, which also means you haven't personally tested it.

Reply:
@chipzoller  I used chatgpt for reference but I thought using secrets  pattern will be helpful to get all resources such as pod and deployment to test type of that ```type: kubernetes.io/service-account-token```. I tested my files of resoure.yaml where I referred from various blogs  https://medium.com/@th3b3ginn3r/understanding-service-accounts-in-kubernetes-e9d2abe19df8 and https://spacelift.io/blog/kubernetes-secrets but I found error regarding this policy.

Reply:
@chipzoller I tried with ```resource.object.type``` it's also not working. I will try with CEL.

Reply:
Please do not use ChatGPT, or any other AI tool for that matter, to tell you what policies to write. Not only is it wrong, but it's just fantasy. Please use the docs, existing samples, and `kubectl explain` to understand the fields available when authoring policies. You do not need to resort to CEL to implement this simple policy.

Reply:
@chipzoller Sure. Thank you for your feedback. I will look into it.